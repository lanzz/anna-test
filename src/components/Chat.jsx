import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';

import Message from './Message';

function Chat({ chat }) {
    const messages = chat ? chat.messages.map((message) => {
        return (
            <Message ts={message.ts} from={message.from} key={message.ts}>
                {message.content}
            </Message>
        );
    }) : undefined;
    return (
        <div className="chat">
            {messages}
        </div>
    );
}

Chat.propTypes = {
    chat: PropTypes.object
};

const mapStateToProps = (state) => {
    return {
        chat: state.selectedChat
    }
};

export default connect(mapStateToProps)(Chat);
