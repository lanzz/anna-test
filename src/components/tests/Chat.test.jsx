import React from 'react';
import { render } from 'enzyme';
import Chat from '../Chat';

it('renders correctly', () => {
    const wrapper = render(
        <Chat />
    );
    expect(wrapper).toMatchSnapshot();
});
