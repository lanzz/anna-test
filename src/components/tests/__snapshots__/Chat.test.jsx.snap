// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  class="chat"
>
  <div
    class="message from-HOST"
  >
    <div
      class="ts"
    >
      Sent on Sun Jan 01 2017 12:34:56 GMT+0000 (GMT)
    </div>
    <div
      class="content"
    >
      Welcome to the chat
    </div>
  </div>
  <div
    class="message from-USER"
  >
    <div
      class="ts"
    >
      Sent on Sun Jan 01 2017 12:34:56 GMT+0000 (GMT)
    </div>
    <div
      class="content"
    >
      hello world
    </div>
  </div>
  <div
    class="message from-HOST"
  >
    <div
      class="ts"
    >
      Sent on Sun Jan 01 2017 12:34:56 GMT+0000 (GMT)
    </div>
    <div
      class="content"
    >
      You said: hello world
    </div>
  </div>
</div>
`;
